server/src/main/java/org/opensearch/search/aggregations/bucket/BestBucketsDeferringCollector.java

```diff
@@ -101,6 +101,7 @@ static class Entry {
     public BestBucketsDeferringCollector(SearchContext context, boolean isGlobal) {
         this.searchContext = context;
         this.isGlobal = isGlobal;
+        this.finished = context.searcher().getLeafContexts().isEmpty();
```

Comment:
I'm trying to decide if it's worth commenting this line with something like "If we have no segments, we're finished before we start". 

Like, I know why you changed it (because Lucene never calls the post-collect method when there are no segments), but I'm wondering if it's sufficiently obvious that "empty implies finished". I kind of think it's obvious, but maybe I have too much context?

Reply:
Added a comment here